hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -474,13 +477,36 @@ private Stream<HoodieInstant> getCommitInstantsToArchive() throws IOException {
               oldestInstantToRetainForClustering.map(instantToRetain ->
                       HoodieTimeline.compareTimestamps(s.getTimestamp(), LESSER_THAN, instantToRetain.getTimestamp()))
                   .orElse(true)
+          ).filter(s ->
+              latestCleanRetainedInstant.map(instant ->
+                      compareTimestamps(s.getTimestamp(), LESSER_THAN, instant))
```

Comment:
I can get the error use case for incremental cleaning, but block the archiving of instants with cleaning is kind of risky, it is better we fix the incremental cleaning procedure.

Reply:
we can fallback to full cleaning if there are archived retained commits.

Reply:
or just use archived timeline to continue incremental clean

Reply:
archived timeline should also be good if the loading does not happen in high frequency.